### PR TITLE
4.13.1

### LINKED ISSUE #85: doesn't compile with clang 16 on macOS
### Comment:

upstream issue: https://gitlab.onelab.info/gmsh/gmsh/-/issues/2637

### PR DESCRIPTION
**Pull request by adrianinsaval**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

closes #93 
closes #85

### CODE REVIEW

**Comment by adrianinsaval** [comment]
what's up with ci?

**Comment by hmaarrfk** [comment]
@conda-forge-admin please restart cis

**Comment by hmaarrfk** [comment]
it was broken just for July 4th-5th

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 MACOSX_SDK_VERSION:  # [osx and x86_64]
-  - 10.12            # [osx and x86_64]
+  - 10.13            # [osx and x86_64]
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:2]: Its probably best to just remove these lines.

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
 occt:
-  - 7.6.3
-  - 7.7.2
\ No newline at end of file
+  - 7.7.2
+  - 7.8.1
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:5]: This typically makes migrations much harder. Do you have a real need for 2 version of OCCT for one of your packages? Or are you trying to be accommodating in a hypothetical case?

Comment (hmaarrfk) [reply recipe/conda_build_config.yaml:5]: For example, the migration https://github.com/conda-forge/gmsh-feedstock/pull/90 failed because of these pins.

**Comment by hmaarrfk** [comment]
PS. I would recommend using
```
        - {{ pin_subpackage("gmsh", exact=True) }}
```

instead of the current pinning since it ensure the build numbers are in sync too (in case you ever need to do some patching)

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (looooo) [reply recipe/conda_build_config.yaml:5]: Current development builds of freecad are still using occt 7.7.2. I guess that's the reason for having occt7.7.2 added here.

**Comment by looooo** [comment]
@conda-forge-admin, please rerender

**Comment by looooo** [comment]
@conda-forge-admin, please rerender

**Comment by looooo** [comment]
@hmaarrfk  Can you review this PR?

**Comment by minrk** [comment]
CI failing due to https://github.com/conda-forge/openexr-feedstock/issues/49 should be fixed in an hour or so

**Comment by isuruf** [comment]
@minrk, please don't merge PRs when you are not a maintainer. If you want to be, send a PR.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -80,7 +81,7 @@ outputs:
     script: install_python.bat  # [win]
     requirements:
       run:
-        - {{ pin_subpackage("gmsh", max_pin="x.x.x") }}
+        - {{ pin_subpackage("gmsh", exact=True) }}
```
[the remark this answers is not in the document]
Comment (isuruf) [reply recipe/meta.yaml:84]: @hmaarrfk is not a maintainer either.

[the remark this answers is not in the document]
Comment (minrk) [reply recipe/meta.yaml:84]: sorry, misread the metadata. I'm working on too many things at once today.

**Pull request merged**